arch/risc-v/include/irq.h

```diff
@@ -703,6 +704,16 @@ irqstate_t up_irq_enable(void);
 int up_cpu_index(void) noinstrument_function;
 #endif /* CONFIG_ARCH_HAVE_MULTICPU */
 
+/****************************************************************************
+ * Schedule acceleration macros
+ ****************************************************************************/
+
+#ifdef CONFIG_RISCV_PERCPU_SCRATCH
+#define up_current_regs() (this_task()->xcp.regs)
+#define up_this_task() (((riscv_percpu_t *)READ_CSR(CSR_SCRATCH))->tcb)
```

Comment:
One solution that would not require exposing riscv internal definitions to the whole world would be to simply make these into normal functions. There is no need to inline the code.

struct tcb_s *up_this_task(void);
void up_update_task(struct tcb_s *tcb);

I prefer you did it this way

Reply:
Another option would be the way you did it originally and simply agree that tcb must be the first element in the percpu structure. This is not unheard of.

Reply:
Current implementation about `up_this_task` is here:
https://github.com/apache/nuttx/blob/1c7d81881c4dfdf1e3eccb82b96c238ce819bd6a/sched/sched/sched.h#L361-L388

Can not define it as a function for now, what's your idea? @hujun260

Reply:
Ok, then it cannot be done as function.